src/odemis/gui/util/img.py

```diff
@@ -860,24 +860,28 @@ def calculate_ticks(value_range, client_size, orientation, tick_spacing):
     vtp_ratio = abs(pixel_space / value_space)  # px/unit
     epsilon = (1 / vtp_ratio) / 10  # "tiny" value: a 10th of a pixel
 
-    num_ticks = pixel_space // tick_spacing
+    # Find the ticks to show so that it looks good for the user. It must have a
+    # constant spacing between the ticks, and this spacing should be a "easy
+    # number": 1, 2, 5, 7.5, or any of these values multiplied by a power of 10.
+    # For example, 150 is not good, but 200 is fine.
+
+    # minimum number of ticks that we want
+    num_ticks = max(2, pixel_space // tick_spacing + 1)
 
-    # set the list of possible value steps
-    spacing_values = [1, 2, 5, 7.5]
+    # minimum tick spacing for getting the number of ticks
+    min_value_step = abs(value_space) / (num_ticks - 1)
 
-    # set the initial value of value_step
-    value_step = abs(value_space) / (num_ticks + 1)
+    # Start with the power of 10 just below
+    power = math.floor(math.log10(min_value_step))
 
-    while value_step and abs(value_space)/value_step > num_ticks:
-        # round down to a power of 10
-        power = int(math.log10(value_step))
-        for step in spacing_values:
-            value_step = step * 10 ** power
-            if abs(value_space) / value_step < num_ticks:
-                break
-    logging.debug("Value step is %s with range %s", value_step, value_space)
+    for step in [1, 2, 5, 7.5, 10]:
```

Comment:
Maybe it is handy to define these 'easy' numbers as a global variable so that a testcase can use the same.
Btw 10 equals 1 * 10 so I don't think you need to also include that number in the list.

Reply:
10 is necessary. That was actually one of the errors in the previous code. As we round down to the lower power of 10 first, if the value is above 7.5, it wouldn't find anything and loop infinitely (and now, it wouldn't loop infinitely, but find 7.5). With 10, we ensure that any value above 7.5 returns 10, which is a nice multiple of 10 :-)

Reply:
Yes, in the previous version, the code kept increasing the `power` (with a while loop) until the value was sufficiently big. So 10 was equivalent to 1, with just the next power. However, I've removed this while loop, as it's safer (makes it impossible to loop infinitely), and it's possible to compute immediately the right `power` anyway. Now, 10 is necessary.